src/xrGame/Weapon.cpp

```diff
@@ -1309,6 +1311,11 @@ void CWeapon::UpdateHUDAddonsVisibility()
 
     //.	return;
 
+    u16 bone_id;
+    bone_id = HudItemData()->m_model->LL_BoneID(wpn_grenade_launcher);
+    if (bone_id == BI_NONE && ShadowOfChernobylMode)
+        wpn_grenade_launcher = WPN_GRENADE_LAUNCHER_SOC;
```

Comment:
We try to minimize usage of `<Game>Mode` variables and add support for features from SOC/CS/COP simultaneously.

Add this to the static shared_str's code block:
```cpp
static shared_str wpn_grenade_launcher_soc = WPN_GRENADE_LAUNCHER_SOC;
```
And then change the code to this:
```suggestion
    u16 bone_id = HudItemData()->m_model->LL_BoneID(wpn_grenade_launcher);
    if (bone_id == BI_NONE)
        bone_id = HudItemData()->m_model->LL_BoneID(wpn_grenade_launcher_soc);
```